### ISSUE TITLE
[Reproducibility] Significant FID discrepancy despite matching LPIPS/PSNR

### ISSUE THREAD

**Issue by TLDBN** [xizaoqu/WorldMem#24]
Hi, thanks for your excellent work on this project!

I am attempting to reproduce your results by retraining the model on 8 A100 GPUs. I followed the provided guidance and used the scripts below.

Since I am using 8 GPUs with a batch size of 4 each (Total Batch Size = 32), I adjusted the configuration by halving the `max_steps` for each stage and doubling the learning rate to maintain training equivalence.

**Training Scripts Used:**
```
# Stage 1
srun python -m main +name=train \
    +diffusion_model_path=/WorldMem_Repro/checkpoints/diffusion_only.ckpt \
    +vae_path=/WorldMem_Repro/checkpoints/vae_only.ckpt \
    +customized_load=true \
    +seperate_load=true \
    +zero_init_gate=true \
    dataset.n_frames=8 \
    dataset.save_dir=WorldMem_Repro/worldmem_minecraft_dataset \
    +dataset.n_frames_valid=700 \
    +dataset.angle_range=110 \
    +dataset.pos_range=2 \
    +dataset.memory_condition_length=8 \
    +dataset.customized_validation=true \
    +dataset.add_timestamp_embedding=true \
    +dataset.wo_updown=true \
    +algorithm.n_tokens=8 \
    +algorithm.memory_condition_length=8 \
    algorithm.context_frames=600 \
    +algorithm.relative_embedding=true \
    +algorithm.log_video=true \
    +algorithm.add_timestamp_embedding=true \
    +algorithm.metrics=[lpips,psnr] \
    experiment.training.checkpointing.every_n_train_steps=2500 \
    experiment.training.max_steps=60000 \
    +output_dir=WorldMem_Repro/reproduce_official_set

# Stage 2
srun python -m main +name=train \
    dataset.n_frames=8 \
    dataset.save_dir=WorldMem_Repro/worldmem_minecraft_dataset \
    +dataset.n_frames_valid=700 \
    +dataset.angle_range=110 \
    +dataset.pos_range=8 \
    +dataset.memory_condition_length=8 \
    +dataset.customized_validation=true \
    +dataset.add_timestamp_embedding=true \
    +dataset.wo_updown=true \
    +algorithm.n_tokens=8 \
    +algorithm.memory_condition_length=8 \
    algorithm.context_frames=600 \
    +algorithm.relative_embedding=true \
    +algorithm.log_video=true \
    +algorithm.add_timestamp_embedding=true \
    +algorithm.metrics=[lpips,psnr] \
    experiment.training.checkpointing.every_n_train_steps=2500 \
    resume=ot7jqmgn \
    +output_dir=WorldMem_Repro/reproduce_official_set \
    experiment.training.max_steps=120000

# Stage 3
srun python -m main +name=train \
    dataset.n_frames=8 \
    dataset.save_dir=WorldMem_Repro/worldmem_minecraft_dataset \
    +dataset.n_frames_valid=700 \
    +dataset.angle_range=110 \
    +dataset.pos_range=8 \
    +dataset.memory_condition_length=8 \
    +dataset.customized_validation=true \
    +dataset.add_timestamp_embedding=true \
    +dataset.wo_updown=false \
    +algorithm.n_tokens=8 \
    +algorithm.memory_condition_length=8 \
    algorithm.context_frames=600 \
    +algorithm.relative_embedding=true \
    +algorithm.log_video=true \
    +algorithm.add_timestamp_embedding=true \
    +algorithm.metrics=[lpips,psnr] \
    experiment.training.checkpointing.every_n_train_steps=2500 \
    resume=ot7jqmgn \
    +output_dir=WorldMem_Repro/reproduce_official_set \
    experiment.training.max_steps=350000

```
**Evaluation**:

I followed the instructions and ran evaluate.sh using the following command on 4 GPUs (generating 4 * 160 videos):
```
python -m main +name=infer \
    experiment.tasks=[test] \
    load=WorldMem/checkpoints/epoch0_step350000.ckpt \
    +customized_load=false \
    dataset.validation_multiplier=1 \
    +dataset.seed=42 \
    dataset.n_frames=8 \
    dataset.save_dir=/share_1/users/bonan_ding/worldmem_data/minecraft \
    +dataset.n_frames_valid=700 \
    algorithm.diffusion.sampling_timesteps=20 \
    +algorithm.memory_condition_length=8 \
    +algorithm.lpips_batch_size=16 \
    +algorithm.log_video=true \
    +algorithm.save_local=true \
    +dataset.customized_validation=true \
    +algorithm.n_tokens=8 \
    algorithm.context_frames=600 \
    experiment.test.batch_size=1 \
    experiment.test.limit_batch=160
```

I then calculated the FID on 50 videos (100 frames each).

Checkpoints | FID score ↓ | LPIPS ↓ | MSE ↓ | PSNR ↑
-- | -- | -- | -- | --
Github Report | 15.13 | 0.1667 | - | 24.01
Official checkpoints | 15.6173 | 0.15762 | 0.01639 | 24.47528
Reproduced (My run) | 22.0626 | 0.16053 | 0.01524 | 24.33882

The FID in my reproduced run is significantly worse than both the reported results and the official checkpoint (22.06 vs ~15.6), even though the other metrics (LPIPS, MSE, PSNR) are very close or identical.

Could you please help me check if there is anything wrong with my training configuration or if I am missing a step that might affect the FID specifically? Thanks!
